eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Gateway/DoctrineDatabase.php

```diff
@@ -1443,6 +1443,81 @@ public function repairBrokenUrlAliasesForLocation(int $locationId)
         }
     }
 
+    /**
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    public function deleteUrlNopAliasesWithoutChildren(): int
+    {
+        $platform = $this->connection->getDatabasePlatform();
+        $queryBuilder = $this->connection->createQueryBuilder();
+
+        // The wrapper select is needed for SQL "Derived Table Merge" issue for deleting
+        $wrapperQueryBuilder = clone $queryBuilder;
+        $selectQueryBuilder = clone $queryBuilder;
+        $expressionBuilder = $queryBuilder->expr();
+
+        $selectQueryBuilder
+            ->select('al1.id AS inner_id')
+            ->from($this->table, 'al1')
+            ->leftJoin(
+                'al1',
+                $this->table,
+                'al2',
+                $expressionBuilder->eq('inner_id', 'al2.parent')
```

Comment:
```suggestion
                $expressionBuilder->eq('al1.id.', 'al2.parent')
```
But please don't use `al1` and `al2`. It's not very meaningful. One of them is parent, the other is child, so there's a lot of possibilities (e.g. `u_parent` and `u` respectively) for naming.